samples/GStreamerWPF/GStreamerD3D11/GStreamerCanvas.xaml.cs

```diff
@@ -0,0 +1,99 @@
+﻿using D3D11Scene;
+using Gst;
+using System;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Interop;
+using System.Windows.Media;
+
+namespace GStreamerD3D11 {
+    public partial class GStreamerCanvas : UserControl, IDisposable {
+
+        private D3DImageEx _D3DImageEx;
+        private D3D11TestScene _D3D11Scene;
+
+        public static DependencyProperty EnabledProperty = DependencyProperty.Register("Enabled", typeof(bool), typeof(GStreamerCanvas), new PropertyMetadata(false, new PropertyChangedCallback(EnabledPropertyChanged)));
+
+        public bool Enabled {
+            get { return (bool)GetValue(EnabledProperty); }
+            set { SetValue(EnabledProperty, value); }
+        }
+
+        public static void EnabledPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            GStreamerCanvas streamComponent = (GStreamerCanvas)d;
+            streamComponent.Reload();
+        }
+
+        public GStreamerCanvas() {
+            InitializeComponent();
+            Loaded += Control_Loaded;
+            Unloaded += Control_Unloaded;
+        }
+
+        private void Control_Unloaded(object sender, RoutedEventArgs e)
+        {
+            Dispose();
+        }
+
+        private void Control_Loaded(object sender, RoutedEventArgs e) {
+            Reload();
+        }
+
+        private void Reload() {
+            if (!Enabled)
+            {
+                return;
+            }
+
+            _D3DImageEx = new D3DImageEx();
```

Comment:
before creating a new object should we check if != null dispose or check if disposed?